libraries/persistent_store/src/driver.rs

```diff
@@ -0,0 +1,630 @@
+    pub fn partial_power_on(
+        mut self,
+        interruption: StoreInterruption,
+    ) -> Result<StoreDriver, (BufferStorage, StoreInvariant)> {
+        self.storage.arm_interruption(interruption.delay);
+        Ok(match Store::new(self.storage) {
+            Ok(mut store) => {
+                store.storage_mut().disarm_interruption();
+                let mut error = None;
+                if let Some(complete) = self.complete {
+                    match StoreDriverOn::new(store, complete.model, &complete.deleted) {
+                        Ok(driver) => return Ok(StoreDriver::On(driver)),
+                        Err((e, x)) => {
+                            error = Some(e);
+                            store = x;
+                        }
+                    }
+                };
+                StoreDriver::On(StoreDriverOn::new(store, self.model, &[]).map_err(
+                    |(rollback, store)| {
+                        let storage = store.into_storage();
+                        match error {
+                            None => (storage, rollback),
+                            Some(complete) => {
+                                let rollback = Box::new(rollback);
+                                let complete = Box::new(complete);
+                                (storage, StoreInvariant::Interrupted { rollback, complete })
+                            }
+                        }
+                    },
+                )?)
+            }
+            Err((StoreError::StorageError, mut storage)) => {
+                storage.corrupt_operation(interruption.corrupt);
+                StoreDriver::Off(StoreDriverOff { storage, ..self })
+            }
+            Err((error, mut storage)) => {
+                storage.reset_interruption();
+                return Err((storage, StoreInvariant::StoreError(error)));
+            }
+        })
+    }
+
+    /// Returns a mapping from delay time to number of modified bits.
+    pub fn delay_map(&self) -> Result<Vec<usize>, (usize, BufferStorage)> {
+        let mut result = Vec::new();
+        loop {
+            let delay = result.len();
+            let mut storage = self.storage.clone();
+            storage.arm_interruption(delay);
+            match Store::new(storage) {
+                Err((StoreError::StorageError, x)) => storage = x,
+                Err((StoreError::InvalidStorage, mut storage)) => {
+                    storage.reset_interruption();
+                    return Err((delay, storage));
+                }
+                Ok(_) | Err(_) => break,
+            }
+            result.push(count_modified_bits(&mut storage));
+        }
+        result.push(0);
+        Ok(result)
+    }
+}
+
+impl StoreDriverOn {
+    /// Provides read-only access to the store.
+    pub fn store(&self) -> &Store<BufferStorage> {
+        &self.store
+    }
+
+    /// Extracts the store.
+    pub fn into_store(self) -> Store<BufferStorage> {
```

Comment:
Because this is not really a conversion. Maybe the naming is not good, I think in the past it was `extract_store` which is maybe better (like the documentation). As you prefer.

Reply:
Renaming is good. If it's not a conversion, it shouldn't be called like one.

Reply:
Done. Also changed `Store::into_storage`.

Note that the convention is not as simple as using `Into` for conversions. For example the following standard functions are conversions that do not use `Into`:
- https://doc.rust-lang.org/alloc/string/struct.String.html#method.into_bytes
- https://doc.rust-lang.org/alloc/vec/struct.Vec.html#method.into_boxed_slice
- https://doc.rust-lang.org/alloc/string/struct.String.html#method.into_boxed_str

But in this case I prefer to not see them as conversions although they could.

Reply:
Noted.